### ISSUE TITLE
Internal debugger error

### ISSUE THREAD

**Issue by luongli** [0xd4d/dnSpy#886]
Hi, I'm reversing some malware written in .Net using dnSpy which is a great tool. However, when I loaded malware into dnSpy and started to debug it, the debugger tab keeps showing the message "internal debuger error". I installed the latest version dnSpy 5.0.5 and .Net 4.7.1, then I tried to set a breakpoint at the cctor of the module but no use, I also read the issues #733 and followed the solution but it didn't work. What I received when I debugged dnSpy itself is the followings:

Exception thrown: 'System.ArgumentException' in mscorlib.dll
Additional information: Illegal characters in path.
Exception thrown: 'System.BadImageFormatException' in Microsoft.CodeAnalysis.dll
Additional  information: Invalid public key.
Exception thrown: 'System.ArgumentException' in Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.dll
Additional information: No module found with MVID '9643bce-30c1.......'

Could you please have a look at this problem and tell me why I'm getting these messages?

**Comment by 0xd4d** [comment]
Since there are no repro steps there's not much I can do.

What's the call stack when an exception is thrown?

**Comment by luongli** [comment]
I decided to dig into dnSpy. After debugging it for a while, I found that the CodeAnalysis library checks the PublicKey flag, if this flag is set, it will load the public key. However, the loading process failed. I opened the malware using CFF explorer to read its meta data header. Under the table .Net Directory\Meta Data Streams\#~\Tables\AssemblyRef, there was one row named iDLaqmCrPyoqarXmNdgSdMwyeYGJA, this assembly had the flags set to 1, which means it has the public key embeded, but the field PublicKeyOrToken was 0. That was a conflict, I modified the value of the flags to 0 and the debugger worked. I think that the row iDLaqmCrPyoqarXmNdgSdMwyeYGJA is added intentionally by the malware developer to defeat the debugger, the malware runs fine with that conflict.

The call_stack of the BadImageFormatException can be found here: https://pastebin.com/Y2Dc1xYM.
